pycparser/_ast_gen.py

```diff
@@ -187,11 +189,38 @@ def _gen_attr_names(self):
 _PROLOGUE_CODE = r'''
 import sys
 
+def _repr(obj):
+    """
+    Get the representation of an object, with dedicated pprint-like format for lists and tuples.
+    """
+    if isinstance(obj, list):
+        return '[' + (',\n '.join((_repr(e).replace('\n', '\n ') for e in obj))) + '\n]'
+    else:
+        return repr(obj) 
 
 class Node(object):
     __slots__ = ()
     """ Abstract base class for AST nodes.
     """
+    def __repr__(self):
+        """ Generates a python representation of the current node
+        """
+        result = self.__class__.__name__ + '('
+        
+        indent = ''
+        separator = ''
+        for name in self.__slots__[:-2]:
```

Comment:
Is there a reason to use `__slots__` directly here? I mean why not just use `chldren` and `attr_names`?

Reply:
Here, `children` would not work for sequences:  `Compound.block_items` would be returned as a list of pairs `('block_item[%d]', <node>)`. 
Infering the parameters from the list of tuples, while possible, would be a lot less efficient (and more error-prone).

As for `attr_names`, this reflects the instance attributes, not the constructor parameters.

To sum it up, using `Compound` as an example:
- `Compound.__slots__` contains `('block_items', 'coord', '__weakref__')`
- `Compound.attr_names` contains `()`
- `Compound.children()` returns `[('block_items[0]', <node_0>), ... , ('block_items[n]', <node_n>)]`